src/ucp/core/ucp_worker.c

```diff
@@ -110,10 +110,14 @@ static ucs_mpool_ops_t ucp_rkey_mpool_ops = {
     kh_int64_hash_func((uintptr_t)(_uct_ep))
 
 
-KHASH_IMPL(ucp_worker_discard_uct_ep_hash, uct_ep_h, char, 0,
+KHASH_IMPL(ucp_worker_discard_uct_ep_hash, uct_ep_h, ucp_request_t*, 1,
            ucp_worker_discard_uct_ep_hash_key, kh_int64_hash_equal);
 
 
+/* Forward declaration */
+static void ucp_worker_discard_cleanup(ucp_worker_h worker);
```

Comment:
why not just define before `ucp_worker_destroy`?

Reply:
there are dependencies on discard callbacks